spec/integration/command_macros_spec.rb

```diff
@@ -46,4 +47,34 @@
     user = repo.users.by_id(user.id).one
     expect(user.name).to eql('Jane')
   end
+
+  it 'allows to pass a block to configure command class' do
+    repo = Class.new(ROM::Repository[:users]) do
+      command(:create) do
+        use :upcase_name
+      end
+    end.new(rom)
+
+    user = repo.create(name: 'Jane')
+
+    persisted_user = repo.users.by_id(user.id).one
+    expect(persisted_user.name).to eql('JANE')
+  end
+
+  it 'allows to pass a block with view syntax' do
+    repo = Class.new(ROM::Repository[:users]) do
+      command :create
+      command update: :by_id do
+        use :upcase_name
+      end
```

Comment:
after working some time with this branch and thinking about hashing/caching of proc-values I tend to agree with you. The only problem with that is that in my code I need not only to extend commands by including modules/plugins but also to configure it in some primitive way. Such as 
```